kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -362,6 +366,28 @@ protected void close() {
                 currentConnectedGroup.remove(clientHost);
             }
 
+            // Try to remove all stored groupID on the metadata store.
+            if (log.isDebugEnabled()) {
+                log.debug("Try to remove all stored groupID on the metadata store. Current connected clientIds: {}",
+                        currentConnectedClientId);
+            }
+            currentConnectedClientId.forEach(clientId -> {
+                String path = groupIdStoredPath + GroupIdUtils.groupIdPathFormat(clientHost, clientId);
+                metadataStore.delete(path, Optional.empty())
+                        .whenComplete((__, ex) -> {
+                            if (ex != null) {
+                                if (ex instanceof MetadataStoreException.NotFoundException) {
```

Comment:
`ex` is a `CompletionException`, we should check `ex.getCause()` rather than `ex` itself.

Reply:
In addition, we can add a debug log when a `NotFoundException`.